docs/index.rst

```diff
@@ -1470,42 +1470,69 @@ Use with tarfile module
 
     Python's `tarfile <https://docs.python.org/3/library/tarfile.html>`_ module supports arbitrary compression algorithms by providing a file object.
 
-    This code encapsulates a ``ZstdTarFile`` class using :py:class:`ZstdFile`, it can be used like `tarfile.TarFile <https://docs.python.org/3/library/tarfile.html#tarfile.TarFile>`_ class:
-
     .. sourcecode:: python
 
         import tarfile
 
-        # when using read mode (decompression), the level_or_option parameter
-        # can only be a dict object, that represents decompression option. It
-        # doesn't support int type compression level in this case.
+        # compression
+        with ZstdFile('archive.tar.zst', mode='w') as _fileobj, tarfile.open(fileobj=_fileobj, mode='w') as tar:
+            # do something
+
+        # decompression
+        with ZstdFile('archive.tar.zst', mode='r') as _fileobj, tarfile.open(fileobj=_fileobj) as tar:
+            # do something
+
+    Alternatively, it is possible to extend the ``Tarfile`` class, except that it supports decompressing ``.tar.zst`` file automatically, as well as adding the following modes: ``r:zst``, ``w:zst`` and ``x:zst``.
+
+    .. sourcecode:: python
+
+        from tarfile import TarFile, CompressionError, ReadError
+
+        class CustomTarFile(TarFile):
+
+            OPEN_METH = {
+                **TarFile.OPEN_METH,
+                'zst': 'zstopen'
+            }
+
+            @classmethod
+            def zstopen(cls, name, mode='r', fileobj=None, level_or_option=None, zstd_dict=None, **kwargs):
+                """Open zstd compressed tar archive name for reading or writing.
+                    Appending is not allowed.
+                """
+                if mode not in ('r', 'w', 'x'):
+                    raise ValueError("mode must be 'r', 'w' or 'x'")
+
+                try:
+                    from pyzstd import ZstdFile, ZstdError
+                except ImportError:
+                    raise CompressionError("pyzstd module is not available") from None
```

Comment:
I think it’s strange to define this class with `zst` in OPEN_METH if the module is not available. I would have the check at top level, or just not have it (expecting people who would copy it to depend on pyzstd for their project).